src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1434,6 +1434,19 @@ def get_store_info(self) -> ServerModel:
         # the one fetched from the global configuration
         model.id = self.get_deployment_id()
 
+        # if ZENML_SERVER_BASE_URL is set as an environment variable, use it
+        # as the base URL for the server
+        base_url = os.environ.get("ZENML_SERVER_BASE_URL")
+
+        # if cloud org ID and tenant ID are set as environment variables, construct
+        # the URL by appending the org ID and tenant ID to the base URL
+        if base_url:
+            model.server_url = base_url
+            cloud_org_id = os.environ.get("ZENML_CLOUD_ORGANIZATION_ID")
+            cloud_tenant_id = os.environ.get("ZENML_CLOUD_TENANT_ID")
```

Comment:
This works, but makes the ZenML OSS code too interlinked with the Cloud API and that makes it difficult to roll out changes independently.

Imagine that in the future, we might change the tenant URL to be constructed differently. All tenants currently deployed at that time will be instantly invalidated and they would all need to be upgraded for the change to take effect.

May I suggest a better solution that keeps things nicely decoupled:

* construct and pass the full `{model.url}/organizations/{cloud_org_id}/tenants/{cloud_tenant_id}"` URL in the `ZENML_SERVER_BASE_URL` env variable instead
* don't use any mention of cloud organization and tenant here

Reply:
@AlexejPenner what do you think about this?

Reply:
I'm not sure I understand what you are suggesting. Are you saying we shouldn't have ORG_ID as a separate property in the sql zen store or as as separate env_var in general?

I believe we will need the org id for event aggregation - I wouldn't want to have to splice this out of some URL from the environment. I also think its fine here to acknoledge on the Tenant side, that something like a ZenML Cloud Org entity exists. However, I don't see why we would need the org_id in the sql zen store

Reply:
I agree with @stefannica that this should probably be passed as the full tenant URL.

For the analytics we still need the organization ID, it would be nice if you could include that as part of this PR in the `ServerConfiguration`

Reply:
I added a `cloud_organization_id` as a variable in the ServerConfiguration. I was leaning towards adding the `metadata` field instead and having the org ID as a key in it but how far would it go in decoupling OSS and cloud?
When we want to access the value for analytics, we would still need to know what that key is called and this would inevitably lead to coupling. Am I missing something? @stefannica

Reply:
@wjayesh if you're still exploring this: you could add a `metadata: Dict[str, str]` field in the `ServerConfiguration` and pass that as an environment variable containing additional cloud specific values e.g. `ZENML_SERVER_METADATA='{"cloud_organization_id: "<org-UUID>"}'. You wouldn't need to know what the key is called on the OSS side if we pass the whole `server_config().metadata` dictionary to the telemetry without caring what's inside it.

Reply:
With a generic `metadata` field in the server settings, we can add whatever values we want in there on the Cloud API side without the OSS code even being aware of it and have those values published in the telemetry, shown in the server information and probably used in other places in the future (e.g. event logging).